lib/utils/extractMagicBlocks.ts

```diff
@@ -44,8 +44,18 @@ export function extractMagicBlocks(markdown: string) {
 /**
  * Restore extracted magic blocks back into a markdown string.
  */
-export function restoreMagicBlocks(replaced: string, blocks: BlockHit[]) {
+export function restoreMagicBlocks(replaced: string, blocks: BlockHit[]) {  
+  let content = replaced;
+
+  // If a magic block is at the start of the document, the extraction token's prepended 
+  // newline will have been trimmed during processing. We need to account for that here
+  // to ensure the token is found and replaced correctly.
```

Comment:
You can see on L34 https://github.com/readmeio/markdown/pull/1276/changes#diff-7cf86c7ad3c32c0baa91e4edc5ede612f869095b4ec0da688d5896b0f03d5861R34 that the token contains a `\n`. We added that here: https://github.com/readmeio/markdown/pull/1246 to ensure the token is parsed at block level.


But the remark serialization step trims that when the token is at the start of the first line in the doc so we need to account for that.